src/lib/timeseries.js

```diff
@@ -0,0 +1,166 @@
+    return timeserie.series.map(serie =>
+      transformTimeserieToTrip({
+        ...serie,
+        geojsonId: timeseries[index]._id
+      })
+    )
+  })
+}
+
+/**
+ * Add aggregates for all timeseries by computing section's datas
+ * @param {array} timeseries - Timeseries to be aggregated
+ * @returns {array} The aggregated timeseries
+ */
+export const computeAggregatedTimeseries = timeseries => {
+  const aggregatedTimeseries = timeseries.map(timeserie => {
+    const serie = timeserie.series[0]
+    let totalSerieCO2 = 0
+    let totalSerieDistance = 0
+    let totalSerieDuration = 0
+    const sections = getSectionsInfo(serie)
+
+    const computedSections = sections.map(section => {
+      const totalCO2 = computeCO2Section(section)
+      totalSerieCO2 += totalCO2
+      totalSerieDistance += section.distance
+      totalSerieDuration += section.duration
+
+      return { ...section, totalCO2 }
+    })
+
+    return {
+      ...timeserie,
+      aggregation: {
+        totalCO2: totalSerieCO2,
+        totalDistance: totalSerieDistance,
+        totalDuration: totalSerieDuration,
+        sections: computedSections
+      }
+    }
+  })
+
+  return aggregatedTimeseries
+}
+
+/**
+ * Computes the total CO2 for all aggregated timeseries
+ * @param {array} aggregatedTimeseries - Aggregated timeseries
+ * @returns {number} The total CO2 of all timeseries
+ */
+export const computeCO2Timeseries = aggregatedTimeseries => {
+  let totalCO2 = 0
+
+  aggregatedTimeseries.forEach(aggregatedTimeserie => {
+    totalCO2 += aggregatedTimeserie.aggregation.totalCO2
+  })
+
+  return totalCO2
+}
+
+/**
+ * Sort modes from aggregated timeseries by CO2 and timeseries count
+ * @param {object} timeseriesByModes - Sorted aggregated timeseries by modes
+ * @returns {object} Sorted aggregated timeseries by CO2
+ */
+export const sortTimeseriesByModesByCO2 = timeseriesByModes => {
+  const pairedTimeseriesByModes = toPairs(timeseriesByModes)
+
+  const unknown = timeseriesByModes[UNKNOWN_MODE]
+  const isUnknownWithNoDatas =
+    unknown.timeseries.length === 0 && unknown.totalCO2 === 0
+
+  const withCO2 = sortBy(
+    pairedTimeseriesByModes.filter(el =>
+      isUnknownWithNoDatas
+        ? el[0] !== UNKNOWN_MODE && el[1].totalCO2 > 0
+        : el[1].totalCO2 > 0
+    ),
+    el => el[1].totalCO2
+  ).reverse()
+
+  const withZeroCO2 = sortBy(
+    pairedTimeseriesByModes.filter(el =>
+      isUnknownWithNoDatas
+        ? el[0] !== UNKNOWN_MODE && el[1].totalCO2 === 0
+        : el[1].totalCO2 === 0
+    ),
+    el => el[1].timeseries.length
+  ).reverse()
+
+  return {
+    ...fromPairs(withCO2),
+    ...fromPairs(withZeroCO2),
+    ...(isUnknownWithNoDatas && { [UNKNOWN_MODE]: unknown })
+  }
+}
+
+/**
+ * Sort aggregated timeseries by modes and CO2
+ * @param {array} aggregatedTimeseries - Aggregated timeseries
+ * @returns {array} Sorted aggregated timeseries by CO2
+ */
+export const sortTimeseriesByModes = aggregatedTimeseries => {
```

Comment:
The function name does not reflect what it does: group aggregated series by mode and sort them by CO2 consumption. It might be clearer like this for instance : `sortTimeseriesByCO2GroupedByMode`